sklearn/_isotonic.pyx

```diff
@@ -62,9 +60,9 @@ def _inplace_contiguous_isotonic_regression(floating[::1] y, floating[::1] w):
             i = k
 
 
-def _make_unique(cnp.ndarray[dtype=floating] X,
-                 cnp.ndarray[dtype=floating] y,
-                 cnp.ndarray[dtype=floating] sample_weights):
+def _make_unique(const floating[:] X,
```

Comment:
I was a little too quick with the suggestion:

```suggestion
def _make_unique(const floating[::1] X,
```

The call-site uses integer indexing which makes a copy, which means `X` is contagious.

https://github.com/scikit-learn/scikit-learn/blob/e2d120981fa8e520a8bec85e9902530b0c7c15b5/sklearn/isotonic.py#L279-L281